File: http/http-server-annotation-processor/src/test/java/ru/tinkoff/kora/http/server/annotation/processor/controller/TestControllerWithResponseEntity.java
package ru.tinkoff.kora.http.server.annotation.processor.controller;

import reactor.core.publisher.Mono;
import ru.tinkoff.kora.http.common.annotation.HttpRoute;
import ru.tinkoff.kora.http.common.annotation.Query;
import ru.tinkoff.kora.http.server.common.HttpServerResponseEntity;
import ru.tinkoff.kora.http.server.common.annotation.HttpController;

import static ru.tinkoff.kora.http.common.HttpMethod.GET;

@HttpController
public class TestControllerWithResponseEntity {

    @HttpRoute(method = GET, path = "/test")
    public HttpServerResponseEntity<String> test(@Query("code") int code) {
        return new HttpServerResponseEntity<>(code, Integer.toString(code));
    }


    @HttpRoute(method = GET, path = "/test2")
    public Mono<HttpServerResponseEntity<String>> mono(@Query("code") int code) {
        return Mono.just(new HttpServerResponseEntity<>(code, Integer.toString(code)));
    }

    @HttpRoute(method = GET, path = "/test3")
    public HttpServerResponseEntity<byte[]> byteArray(@Query("code") int code) {
        return new HttpServerResponseEntity<>(code, new byte[]{});
    }
}

File: http/http-server-symbol-processor/src/test/kotlin/ru/tinkoff/kora/http/server/symbol/processor/controllers/TestControllerWithResponseEntity.kt
package ru.tinkoff.kora.http.server.symbol.processor.controllers

import kotlinx.coroutines.delay
import ru.tinkoff.kora.http.common.HttpMethod
import ru.tinkoff.kora.http.common.annotation.HttpRoute
import ru.tinkoff.kora.http.common.annotation.Query
import ru.tinkoff.kora.http.server.common.annotation.HttpController
import ru.tinkoff.kora.http.server.common.HttpServerResponseEntity

@HttpController
open class TestControllerWithResponseEntity {
    @HttpRoute(method = HttpMethod.GET, path = "/test")
    open fun test(@Query("code") code: Int): HttpServerResponseEntity<String> {
        return HttpServerResponseEntity(code, code.toString())
    }

    @HttpRoute(method = HttpMethod.GET, path = "/test2")
    open suspend fun suspended(@Query("code") code: Int): HttpServerResponseEntity<String> {
        delay(2)
        return HttpServerResponseEntity(code, code.toString())
    }

    @HttpRoute(method = HttpMethod.GET, path = "/test3")
    open fun byteArray(@Query("code") code: Int): HttpServerResponseEntity<ByteArray> {
        return HttpServerResponseEntity(code, byteArrayOf())
    }
}
